README.md

```diff
@@ -1,2 +1,401 @@
+
+
+# Module Overview
+
+The Zephyr Project is a scalable real-time operating system (RTOS) supporting
+multiple hardware architectures, optimized for resource constrained devices,
+and built with security in mind.
+
+The Zephyr build system supports code tree "plug-ins" called *modules*.
+See the [Zephyr Modules (External projects)][5] for a high-level overview
+of the build system module mechanism.
+
+
+## Scope
+
+The Zephyr eXample Application (ZXA) module is supplementary to the Zephyr
+repository.  This provides:
+
+1. A reference implementation of using the Zephyr module mechanism.
+
+1. A repository for holding example code and config files outside of the
+   Zephyr repository, including:
+
+   - `west` manifest
+   - projects/samples/tests
+   - applications/subsystems/drivers/interface declarations
+   - device tree config files
+   - boards
+   - scripts
+
+1. A test case for CI to monitor Zephyr build system support for modules.
+
+
+   _WARNING:_ The Zephyr repository is intentionally
+   **independent of and unaware of this repository**.
+
+
+# Getting Started
+
+## Creating a New Module
+
+This module is designed to be a boilerplate for other modules simply by:
+
+1. cloning
+
+1. renaming module names and abbreviations to reference your module:
+
+  - 'eXample Application'
+  - 'example-application'
+  - 'ZXA'
+  - 'zxa'
+
+1. stripping out any unnecessary content (e.g. no drivers or subsystems)
+
+  - If you don't have any non-sample, non-test projects:
+
+    1. remove the `zephyr/projects/` subdirectory
+    1. In `zephyr/modules.yml` remove the reference to `zephyr/projects`
+
+
+## Creating a Free-Standing Manifest Repository
+
+This example module is also an appropriate template for creating a
+manifest-only repository because, like manifest modules, they may
+contain the following:
+
+- multiple manifest files
+
+- documentation describing the scope and purpose of the workspace
+  configured by the manifest
+
+- configuration settings for CI pipelines
+
+- scripts unique to the breadth of the workspace
+
+To strip this repository down to be a manifest-only repository:
+
+  1. Remove all files and subdirectories referenced by `zephyr/module.yml`
+  1. Remove the `zephyr/include` subdirectory
+  1. Remove `zephyr/module.yml`
+
+
+## Checking Out Code
+
+The Zephyr module ZXA may be cloned and added to the Zephyr build system
+in exactly the same way as any other repository.
+
+Once the Zephyr development environment has been set up (see [Zephyr Getting Started Guide][3]), then follow instructions here to:
+
+- Get the source code with Zephyr module ZXA as the `west`
+  manifest repository
+- Build and run projects
+
+
+## Get Zephyr and Install Python Dependencies
+
+Clone Zephyr module ZXA and its dependencies into a new `west`
+workspace named :file:`zp`. You'll also install additional Python
+dependencies.
+
+### Ubuntu
```

Comment:
I think we should avoid OS-specific instructions here.
If you limit yourself to:

```
west init -m https://github.com/zephyrproject-rtos/example-application
cd example-application
west update
```
that should be more than enough?
`--mr main` is not needed, it's the default
`--mf zephyr/west.yml` is not needed if we move `west.yml` to the root, where it IMHO belongs
`source zephyr-env.sh` is not needed if you have done `west zephyr-export` and have a single zephyr installation.
`pip3 install ...` this is assuming that you have additional python packages, but that is an assumption. Let's remove that too.
`west config build.dir-fmt ${ZEPHYR_BASE}/../build/{source_dir}/{board}`: this is a user preference, let's not choose for them.

Reply:
Again, let's start small, build on that.

Reply:
>  `--mr main` is not needed, it's the default

Master is the default actually.

Reply:
Otherwise, fully agree.

Reply:
@carlescufi, @mbolivar-nordic  In order for this example to be suitable for extending an existing repository and to demonstrate the `--mf <manifestfile>` mechanism exists, the manifest file must be within the `zephyr` folder. It will be **much** easier to get other projects to extend their support to include Zephyr if we keep everything Zephyr-specific out of the top-level directory.